String parse method added

s4kibs4mi opened the pull request:
Method to parse json value from raw string data added.
As there's a parse() method having parameter String this method's parameter taken as StringBuilder.

cbeust commented:
Could you split this into two PR's, one of which contains only the code changes and the other one contains the reformatting, reindenting, space trimming, etc...?

As it is, it's a bit hard to find what you actually changed.

Thanks.

s4kibs4mi commented:
Sure.. I'm splitting into two PRs..But what I actually did..I have added a method in **Parser.kt**
so that anyone can parse json directly from raw string data, which is missing in the library.

```
fun parse(rawValue: StringBuilder): Any? {
    val inputStream: InputStream = ByteArrayInputStream(rawValue.toString().toByteArray())
    return parse(inputStream)
}
```
if it's ok then please merge the PR.
And I have added the example in **Readme.md** file.
which is,

```
val parser: Parser = Parser()
val stringBuilder: StringBuilder = StringBuilder("{\"name\":\"Sakib Sami\", \"age\":23}")
val json: JsonObject = parser.parse(stringBuilder) as JsonObject
println("Name : ${json.string("name")}, Age : ${json.int("age")}")
```
Result :
```
Name : Sakib Sami, Age : 23
```

cbeust commented:
Yes, I suspected your change should be just a few lines but it was drowned in all the other changes :-)

Let's close this PR and you can open a new PR with just that change, which I'll be happy to merge.

Thanks!
